journals/apps/journals/models.py

```diff
@@ -314,6 +315,17 @@ def generate_basket_url(self, sku):
         ))
         return basket_url
 
+    def update_related_objects(self, deactivate=False):
+        data = {
+            "status": "active" if not deactivate else "inactive",
+            "card_image_url": self.site.root_url + self.card_image.file.url,
+            "title": self.title,
+            "full_description": self.long_description,
+            "short_description": self.short_description,
+        }
+        discovery_journal_api_client = self.site.siteconfiguration.discovery_journal_api_client
+        discovery_journal_api_client.journals(self.journal.uuid).patch(data)
```

Comment:
Actually, this is a problem. During the mgmt command, the wagtail pages are created first before the journal data is pushed to discovery. Your .patch() is failing because the journal with specified uuid does not yet exist:
```
root@00334e2d7247:/edx/app/journals/journals# ./manage.py publish_journals --create "Boo-123" --org "edx" --price "200.00" --access_length "180"
2018-04-12 03:17:47,974 INFO 72 [elasticsearch] base.py:83 - PUT http://journals.elasticsearch:9200/wagtail__wagtailcore_page/wagtailcore_page_journals_journalaboutpage/wagtailcore_page%3A53 [status:201 request:0.018s]
2018-04-12 03:17:47,976 INFO 72 [wagtail.core] models.py:503 - Page created: "Boo-123 About Page" id=53 content_type=journals.JournalAboutPage path=/mitxpro-index-page/boo-123-about-page/
2018-04-12 03:17:48,030 INFO 72 [elasticsearch] base.py:83 - PUT http://journals.elasticsearch:9200/wagtail__wagtailcore_page/wagtailcore_page_journals_journalaboutpage/wagtailcore_page%3A53 [status:200 request:0.016s]
2018-04-12 03:17:48,039 INFO 72 [wagtail.core] models.py:685 - Page edited: "Boo-123 About Page" id=53 revision_id=52
2018-04-12 03:17:48,078 INFO 72 [elasticsearch] base.py:83 - PUT http://journals.elasticsearch:9200/wagtail__wagtailcore_page/wagtailcore_page_journals_journalaboutpage/wagtailcore_page%3A53 [status:200 request:0.016s]
Traceback (most recent call last):
  File "./manage.py", line 15, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/edx/app/journals/journals/journals/apps/journals/management/commands/publish_journals.py", line 160, in handle
    journal = self._handle_create(options)
  File "/edx/app/journals/journals/journals/apps/journals/management/commands/publish_journals.py", line 45, in _handle_create
    journal_about_page = self._update_wagtail_pages(journal, create=True, publish=publish)
  File "/edx/app/journals/journals/journals/apps/journals/management/commands/publish_journals.py", line 115, in _update_wagtail_pages
    journal_about_page.save_revision().publish()
  File "/usr/local/lib/python3.6/site-packages/wagtail/wagtailcore/models.py", line 1524, in publish
    page_published.send(sender=page.specific_class, instance=page.specific, revision=self)
  File "/usr/local/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 193, in send
    for receiver in self._live_receivers(sender)
  File "/usr/local/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 193, in <listcomp>
    for receiver in self._live_receivers(sender)
  File "/edx/app/journals/journals/journals/apps/journals/handlers.py", line 16, in about_page_pub_receiver
    journal_about_age.update_related_objects()
  File "/edx/app/journals/journals/journals/apps/journals/models.py", line 333, in update_related_objects
    discovery_journal_api_client.journals(self.journal.uuid).patch(data)
  File "/usr/local/lib/python3.6/site-packages/slumber/__init__.py", line 171, in patch
    resp = self._request("PATCH", data=data, files=files, params=kwargs)
  File "/usr/local/lib/python3.6/site-packages/slumber/__init__.py", line 101, in _request
    raise exception_class("Client Error %s: %s" % (resp.status_code, url), response=resp, content=resp.content)
slumber.exceptions.HttpNotFoundError: Client Error 404: http://edx.devstack.discovery:18381/journal/api/v1/journals/6303f0c8-00c4-424e-9a27-71ff52454688/
```